### DISCUSSION TITLE
Force delete records

### DISCUSSION THREAD

**Discussion by Nazariy** [phalcon/cphalcon#15670] [Q&A]
While Phalcon allows to soft delete records I'm wandering what is a best practice to force delete soft deleted records, is there a specific method for that or it has to be implemented via SQL query?

**Comment by quasipickle** [comment]
I don't think there is a method.  You could probably write a `forceDelete()` method on your base class so you don't need to build an SQL query every time.

I wonder, though, what the use case is for soft deleting records at one time, but then actually deleting at a later time?  Presumably you could just set a column value rather than calling `delete()` - that's all soft deleting does anyway.  This approach would leave `delete()` alone and allow you to use it as intended.

**Reply by Nazariy** [reply]
Some common scenarios for soft delete usage is to be able to recover records that customer have deleted by accident and be able to remove them in batch after month or manually, similar to what Google does with spam folder.

So far I wasn't able to implement this feature within same object as it seems there is no functionality available allowing to detach model behaviour. The solution I came with is to extend existing model and implement soft delete behaviour there , so I can invoke objects based on current ACL state.

It would be nice to have more control over object behaviours and be on same page with other popular frameworks.

**Reply by quasipickle** [reply]
Just don't add the `softDelete` behaviour to the model.  The `SoftDelete()` behaviour is for models that will never be actually deleted - your situation is different.

Write your own `softDelete()` method that sets a column then saves the model.  Then, when you don't want to _actually_ delete in your app, just call `$SomeModel->softDelete()`.  Calling `$SomeModel->delete()` would then still delete the row.

If you're doing batch jobs - an ORM probably isn't the best method.  Better to use raw SQL for those cases.
